src/core/dev/hw_queue_tx.cpp

```diff
@@ -498,28 +493,19 @@ void hw_queue_tx::init_queue()
     m_sq_wqe_hot->eseg.cs_flags = XLIO_TX_PACKET_L3_CSUM | XLIO_TX_PACKET_L4_CSUM;
 
     hwqtx_logfunc("%p allocated for %d QPs sq_wqes:%p sq_wqes_end: %p and configured %d WRs "
-                  "BlueFlame: %p buf_size: %d offset: %d",
+                  "BlueFlame: %p",
                   m_mlx5_qp.qp, m_mlx5_qp.qpn, m_sq_wqes, m_sq_wqes_end, m_tx_num_wr,
-                  m_mlx5_qp.bf.reg, m_mlx5_qp.bf.size, m_mlx5_qp.bf.offset);
+                  m_mlx5_qp.bf.reg);
 }
 
 void hw_queue_tx::init_device_memory()
 {
     /* This limitation is done because of a observation
      * that dm_copy takes a lot of time on VMs w/o BF (RM:1542628)
      */
-    if (m_p_ib_ctx_handler->get_on_device_memory_size() > 0) {
-        if (m_db_method == MLX5_DB_METHOD_BF) {
-            m_dm_enabled =
-                m_dm_mgr.allocate_resources(m_p_ib_ctx_handler, m_p_ring->m_p_ring_stat.get());
-
-        } else {
-#if defined(DEFINED_IBV_DM)
-            VLOG_PRINTF_ONCE_THEN_DEBUG(
-                VLOG_WARNING,
-                "Device Memory functionality is not used on devices w/o Blue Flame support\n");
-#endif /* DEFINED_IBV_DM */
-        }
+    if (m_p_ib_ctx_handler->get_on_device_memory_size() > 0 && is_bf(m_p_ib_ctx_handler->get_ibv_context())) {
+        m_dm_enabled =
+            m_dm_mgr.allocate_resources(m_p_ib_ctx_handler, m_p_ring->m_p_ring_stat.get());
```

Comment:
Do we know if device memory can be used when BF is not?

Reply:
From the XLIO perspective, we allocate device memory explicitly with ibv_alloc_dm(). Therefore an issue similar to BF duplicates is unlikely.
Also, device memory is independent from the doorbell method. According to the comment, it's disabled for devices without blueflame due to some practical observations.